xarray/testing/strategies.py

```diff
@@ -0,0 +1,358 @@
+from collections.abc import Hashable, Mapping, Sequence
+from typing import Any, Protocol, Union
+
+import hypothesis.extra.numpy as npst
+import hypothesis.strategies as st
+import numpy as np
+
+import xarray as xr
+from xarray.core.types import T_DuckArray
+
+__all__ = [
+    "numeric_dtypes",
+    "names",
+    "dimension_names",
+    "dimension_sizes",
+    "attrs",
+    "variables",
+]
+
+
+class ArrayStrategyFn(Protocol):
+    def __call__(
+        self,
+        *,
+        shape: Union[tuple[int, ...], None] = None,
+        dtype: Union[np.dtype, None] = None,
+        **kwargs,
+    ) -> st.SearchStrategy[T_DuckArray]:
+        ...
```

Comment:
@Illviljan @headtr1ck if either of you have time I would appreciate some help with getting this typing to work

Reply:
I think I fixed this, though it required changing the `T_DuckArray` type to be covariant. See https://stackoverflow.com/questions/67404067/how-to-use-typevar-for-input-and-output-of-multiple-generic-protocols-in-python